extension/src/popup/components/identicons/KeyIdenticon/index.tsx

```diff
@@ -15,41 +15,76 @@ interface IdenticonWrapperElProps {
 }
 
 interface KeyIdenticonProps extends IdenticonWrapperElProps {
+  allowCopy?: boolean;
   publicKey: string;
+  iconSide?: "left" | "right";
+  keyTruncationAmount?: number;
 }
 
 export const KeyIdenticon = ({
+  allowCopy = false,
   publicKey = "",
   isSmall = false,
   customSize,
+  keyTruncationAmount,
+  iconSide = "left",
   ...props
 }: KeyIdenticonProps) => {
-  const shortPublicKey = truncatedPublicKey(publicKey);
   const customStyle = {
     ...(isSmall
       ? {
           // eslint-disable-next-line
           "--Icon-padding": "0.2rem",
           // eslint-disable-next-line
           "--Icon-dimension": "1.5rem",
+          marginRight: iconSide === "left" ? "0.5rem" : 0,
+          marginLeft: iconSide === "right" ? "0.5rem" : 0,
         }
-      : {}),
+      : {
+          marginRight: !allowCopy ? "0.5rem" : 0,
+        }),
     ...(customSize
       ? {
           // eslint-disable-next-line
           "--Icon-padding": customSize.padding,
           // eslint-disable-next-line
           "--Icon-dimension": customSize.dimension,
+          marginRight: iconSide === "left" ? "0.5rem" : 0,
+          marginLeft: iconSide === "right" ? "0.5rem" : 0,
         }
-      : {}),
+      : {
+          marginRight: !allowCopy ? "0.5rem" : 0,
+        }),
   } as React.CSSProperties;
 
   return (
     <div className="KeyIdenticon">
-      <div className="KeyIdenticon--icon" style={customStyle}>
-        <IdenticonImg publicKey={publicKey} />
-      </div>
-      <span {...props}>{shortPublicKey}</span>
+      {iconSide === "left" && (
+        <div className="KeyIdenticon--icon" style={customStyle}>
+          <IdenticonImg publicKey={publicKey} />
+        </div>
+      )}
+      {allowCopy ? (
+        <CopyText textToCopy={publicKey}>
+          <span {...props} className="KeyIdenticon--key">
+            <div className="CopyContractId">
```

Comment:
should this just be a custom component `<CopyContractId value= {truncateString(publicKey, keyTruncationAmount)} />` to avoid repeating the jsx/styles?

Reply:
Or if there are too many variants, maybe you could create a mixin or something to reuse the relevant styles. My concern is there's a lot of nested `.CopyContractId`'s which could lead to naming collisions and messiness in the future

Reply:
I made it more generic since there are already a few places where we need it for other string types, so now we have `<CopyValue />`. There are still a few places where we need to tweak the styles for it inline for certain layouts but hopefully this cleans it up a bit. Done in 4b135e5b899b32ec2040d54ce0438b559fe3b135